plugins/import/import.ts

```diff
@@ -73,21 +83,67 @@ async function getEtherscanApiData(network: string, address: string, apiKey: str
   };
 }
 
-async function getContractCreationCode(network: string, address: string) {
-  let url = `${await getEtherscanUrl(network)}/address/${address}#code`;
-  let result = <string>await get(url, {});
-  let regex = /<div id='verifiedbytecode2'>[\s\r\n]*([0-9a-fA-F]*)[\s\r\n]*<\/div>/g;
-  let matches = [...result.matchAll(regex)];
+async function scrapeContractCreationCodeFromEtherscan(network: string, address: string) {
+  const url = `${getEtherscanUrl(network)}/address/${address}#code`;
+  debug(`Attempting to scrape Contract Creation code at ${url}`);
+  const result = <string>await get(url, {});
+  const regex = /<div id='verifiedbytecode2'>[\s\r\n]*([0-9a-fA-F]*)[\s\r\n]*<\/div>/g;
+  const matches = [...result.matchAll(regex)];
   if (matches.length === 0) {
     if (result.match(/request throttled/i) || result.match(/try again later/i)) {
       throw new Error(`Request throttled: ${url}`);
     } else {
       throw new Error(`Failed to pull deployed contract code from Etherscan: ${url}`);
     }
   }
+  debug(`Scraping successful for ${url}`);
   return matches[0][1];
 }
 
+function paramString(params: Object) {
+  return Object.entries(params).map(([k,v]) => `${k}=${v}`).join('&');
+}
+
+async function pullFirstTransactionForContract(network: string, address: string) {
+  const params = {
+    module: "account",
+    action: 'txlist',
+    address,
+    startblock: 0,
+    endblock: 99999999,
+    page: 1,
+    offset: 10,
+    sort: 'asc',
+    apikey: getEtherscanApiKey(network)
+  };
+  const url = `${getEtherscanApiUrl(network)}?${paramString(params)}`;
+
+  debug(`Attempting to pull Contract Creation code from first tx at ${url}`);
```

Comment:
this dumps our various API keys into the logs ☹️ . i should probably remove that param before logging

Reply:
removed